### ISSUE TITLE
SchemaError when selecting columns

### ISSUE THREAD

**Issue by universalmind303** [GlareDB/glaredb#1406]
## Context
When selecting column names with capitalizations, it appears i need to wrap them in quotes. 

```sql
-- This fails
select VendorID from parquet_scan('https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2023-01.parquet') limit 1;
-- this works
select "VendorID" from parquet_scan('https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2023-01.parquet') limit 1;
```

## Expected
I am able to select the column name without quoting it

## Actual
i need to wrap the column name in quotes

## Impact
confusing error & does not comply with sql standards.

**Comment by greyscaled** [comment]
Seeing this elsewhere too:

![VendorID](https://github.com/GlareDB/glaredb/assets/11184711/37741de2-e20b-4834-a240-85a776b791e3)

**Comment by scsmithr** [comment]
This follows postgres naming conventions. Case-sensitive column names need to be quoted.

Also Grey the font looks weird in your screenshot. Looks like the cursor is off a bit?

**Comment by greyscaled** [comment]
> This follows postgres naming conventions. Case-sensitive column names need to be quoted.
> 
> Also Grey the font looks weird in your screenshot. Looks like the cursor is off a bit?

Yea I hit it once then not again on refresh.

This has to do with fonts, Monaco and how Monaco computes pixels. I need to dig into it more deeply since I've hit it more than once.

**Comment by universalmind303** [comment]
> This follows postgres naming conventions. Case-sensitive column names need to be quoted.

while case sensitivity can be enforced by the use of quotes, it isn't the default. 

this is completely valid sql. 

```sql
CREATE TABLE test (
    VendorId INT PRIMARY KEY,
    name VARCHAR(255)
);

INSERT INTO test(vendorid, name) 
VALUES 
(1, 'foo'),
(2, 'bar');

select vendorid from test;
```

however, **the case sensitivity is preserved if it is created using quotes**
```sql
CREATE TABLE test (
    "VendorId" INT PRIMARY KEY,
    name VARCHAR(255)
);

--- errors as 'vendorid' does not exist
INSERT INTO test(vendorid, name) 
VALUES 
(1, 'foo'),
(2, 'bar');
```


so i guess it's not necessarily an error, but a matter of preference if external files such as parquet should have the column names auto quoted to preserve case sensitivity, or if it should only quote columns with non alphanumeric characters.

**Comment by vrongmeal** [comment]
Yup, this is intentional. I think we should keep it this way. It's simpler to argue for (and makes sense to me).
